### PR TITLE
Fix space removing in modem ATD command

### PR DESCRIPTION
**Pull request by NicknineTheEagle**
This snippet removing trailing spaces from ATD command line was not actually working.

### CODE REVIEW

**Comment by kcgen** [comment]
@NicknineTheEagle , can you show examples inside dosbox and/or a terminal program how this problematic code currently behaves versus how it should behave (perhaps with output from this PR)?

**Comment by NicknineTheEagle** [comment]
Easy to check, just open any terminal program type in "atdt bbs.dmine.net   ". Right now, you'll get this:
```
SERIAL: Port 1 command sent to modem: ->ATDT BBS.DMINE.NET   <-
SERIAL: Port 1 connecting to host BBS.DMINE.NET    port 23.
SERIAL: Port 1 failed to connect.
```
With the fix, trailing spaces will get trimmed and the argument will get parsed properly:
```
SERIAL: Port 1 command sent to modem: ->ATDT BBS.DMINE.NET   <-
SERIAL: Port 1 connecting to host BBS.DMINE.NET port 23.
SERIAL: Port 1 modem response: CONNECT 57600.
```

File: src/hardware/serialport/softmodem.cpp [hunk 1]
```diff
@@ -489,7 +489,7 @@ void CSerialModem::DoCommand()
 				foundstr++;
 
 			char* helper = foundstr;
-			helper+=strlen(foundstr);
+			helper += strlen(foundstr) - 1;
```
Comment (dreamer) [inline src/hardware/serialport/softmodem.cpp:492]: How about replacing lines 491-496 with:
```C++
rtrim(foundstr);
```
?

This way the trimming algorithm would remove all whitespace characters, and not only spaces.

**Review by dreamer — APPROVED** [review]
@NicknineTheEagle trimming this way is even better :)

Thank you!

**Pull request merged**